### DISCUSSION TITLE
Getting the different wallet address on different applications.

### DISCUSSION THREAD

**Discussion by KetulCodiste** [Web3Auth/Web3Auth#456] [Help]
I was going through the documentation of web3Auth. web3Auth have provided the examples for integrating the web3 in different frameworks and blockchains. So I copy-pasted 3 examples in my local machine. 1st one is React js, 2nd one is polygon using react and 3rd one from the demo section of git repo itself(react). In all these demo, whenever I try to connect with the wallet, I am getting the different wallet addresses. For example, if I login with magic link which web3Auth provides in email, then all the 3 demos gives different account address. So for the same email address, I am getting 3 different account addresses in 3 different demos. Likewise, if I login with google, I am facing the same issue here as well. And the wallet address I am getting from google login is different from the wallet address that I am getting through the magic link. So, summarising, for the same email address, I am getting different addresses in different demos, and different addresses for different login mode.

**Comment by himanshuchawla009** [comment]
Web3Auth accounts are scoped to applications (clientId).  All those demos must be using different clientId and hence you are getting different accounts for same login methods. 

You can get your application clientId by registering your project at https://dashboard.web3auth.io.

Other reasons can be:-

- Different verifiers being used
- Different client ids
- Different social logins (email passwordless and google is not same)
- Torus wallet accounts are different from web3auth accounts.

**Reply by KetulCodiste** [reply]
Yes, you are right. The clientId was different in my different demos. My bad.
But in one of the demo, having the same clientId, the wallet address changed suddenly. It was like, I made a successful transaction, logged out to check the whole flow of the demo again, and when I logged in, the wallet address was changed.

**Reply by Oberhaim** [reply]
thats a nightmare :(

Having the same issue. Client ID is the same, login method is the same but still I get different addresses
